src/explore-education-statistics-frontend/src/middleware/pages/redirectPages.ts

```diff
@@ -0,0 +1,62 @@
+import redirectService, {
+  Redirects,
+  RedirectType,
+} from '@frontend/services/redirectService';
+import type { NextRequest } from 'next/server';
+import { NextResponse } from 'next/server';
+
+interface StoredRedirects {
+  redirects: Redirects;
+  fetchedAt: number;
+}
+
+let storedRedirects: StoredRedirects | undefined;
+
+const redirectPaths = {
+  methodologyRedirects: '/methodology',
+  publicationRedirects: '/find-statistics',
+};
+
+const cacheTime = 300000;
+
+export default async function redirectPages(request: NextRequest) {
+  const shouldRefetch =
```

Comment:
Do you actually need that first `|| (storedRedirects?.fetchedAt` check if storedRedirects is defined - is there a way for fetchedAt to be falsy at that point?